src/main/java/frc/robot/commands/vision/TestVision2.java

```diff
@@ -0,0 +1,113 @@
+package frc.robot.commands.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.Vision;
+
+public class TestVision2 extends CommandBase{
+  private Drivetrain m_drive;
+  private Vision m_vision;
+  private double m_encoderStart;
+  private Pose2d m_startPose;
+  private double m_encoderPosition;
+  private Pose2d m_currentPose = null;
+  private int m_endCounter = 0;
+  private double m_speed;
+  private int m_direction = 1;
+  private boolean m_turned = false;
+  private double m_distanceToMove;
+  private double m_closest;
+
+  //How many frames it has to not see anything to end the command
+  private static final int endDelay = 5;
+
+  /**
+   * Moves the robot a certain distance and then moves it back
+   * @param speed What speed the robot should move at
+   * @param distance How far it should move
+   * @param drive The drivetrain
+   */
+  public TestVision2(double speed, double distance, Drivetrain drive, Vision vision){
+    addRequirements(drive);
+    m_drive=drive;
+    m_vision=vision;
+    m_speed=speed;
+    m_distanceToMove=distance;
+  }
+
+  private double getDist(){
+    return 
+      m_drive.m_modules[0].getDrivePosition()/4+
+      m_drive.m_modules[1].getDrivePosition()/4+
+      m_drive.m_modules[2].getDrivePosition()/4+
+      m_drive.m_modules[3].getDrivePosition()/4;
+  }
+
+  /**
+   * Initializes the command
+   */
+  @Override
+  public void initialize(){
+    m_encoderStart=getDist();
+    m_startPose=m_vision.getPose2d(m_currentPose, m_drive.getPose());
+    m_direction=1;
```

Comment:
why dose  m_direction exist when you have m_turned or vice versa. you even use m_direction to check if it turned for one rather than useing m_turrned.